fedimint/src/setup/mod.rs

```diff
@@ -0,0 +1,231 @@
+mod configgen;
+
+use std::net::SocketAddr;
+use std::path::{Path, PathBuf};
+use std::sync::{Arc, RwLock, RwLockWriteGuard};
+
+use askama::Template;
+use axum::extract::{Extension, Form};
+use axum::response::Redirect;
+use axum::{
+    routing::{get, post},
+    Router,
+};
+use fedimint_core::config::ClientConfig;
+use http::StatusCode;
+use qrcode_generator::QrCodeEcc;
+use rand::rngs::OsRng;
+use serde::Deserialize;
+use tokio::sync::mpsc::Sender;
+
+use crate::setup::configgen::configgen;
+use crate::ServerConfig;
+use mint_client::api::WsFederationConnect;
+
+fn run_fedimint(state: &mut RwLockWriteGuard<State>) {
+    let sender = state.sender.clone();
+    tokio::task::spawn(async move {
+        // trying to trigger the receive message
+        sender.send(()).await.expect("failed to send over channel");
+    }); // FIXME: it won't let me await this
+    state.running = true;
+}
+
+#[derive(Deserialize, Debug, Clone)]
+#[allow(dead_code)]
+pub struct Guardian {
+    name: String,
+    connection_string: String,
+}
+
+#[derive(Template)]
+#[template(path = "home.html")]
+struct HomeTemplate {
+    running: bool,
+    can_run: bool,
+}
+
+async fn home(Extension(state): Extension<MutableState>) -> HomeTemplate {
+    let state = state.read().unwrap();
+    let can_run = Path::new(&state.cfg_path.clone()).is_file() && !state.running;
+    HomeTemplate {
+        running: state.running,
+        can_run,
+    }
+}
+
+#[derive(Template)]
+#[template(path = "dealer.html")]
+struct DealerTemplate {
+    guardians: Vec<Guardian>,
+}
+
+async fn dealer(Extension(state): Extension<MutableState>) -> DealerTemplate {
+    DealerTemplate {
+        guardians: state.read().unwrap().guardians.clone(),
+    }
+}
+
+async fn add_guardian(
+    Extension(state): Extension<MutableState>,
+    Form(form): Form<Guardian>,
+) -> Result<Redirect, (StatusCode, String)> {
+    state.write().unwrap().guardians.push(Guardian {
```

Comment:
Let's parse the connection string here and display an error if it's not valid.

If it is, store `PublicKey` and `SocketAddr` on `Guardian`, instead of storing the unparsed connection string as we do now.